crates/ruff_python_parser/tests/fixtures.rs

```diff
@@ -230,6 +233,37 @@ impl std::fmt::Display for CodeFrame<'_> {
     }
 }
 
+/// Verifies that:
+/// * the ranges are strictly increasing when loop the tokens in insertion order
+/// * all ranges are within the length of the source code
+fn validate_tokens(tokens: &[Token], source_length: TextSize, test_path: &Path) {
+    let mut token_iter = tokens.iter();
+    let mut previous = token_iter.next();
+
+    for token in token_iter {
+        assert!(
+            token.end() <= source_length,
+            "{path}: Token range exceeds the source code length. Token: {token:#?}",
+            path = test_path.display()
+        );
```

Comment:
Nit: This check won't be executed for `previous` if this is a single token program.

You may want to change the logic to use a lookahead instead (`peekable()` or change the loop to iterate while `current` is `Some`)

Reply:
Good catch. I don't it's even required to do that as previous is an `Option`.